### PR TITLE
[NFC][sanitizer_symbolizer]Add StackTracePrinter class

### PR DESCRIPTION
**Pull request by avillega**
Introduce a new virtual class StackTracePrinter and an implementation
FormattedStackTracePrinter in preparation of enabling symbolizer markup
for linux. 
This change allows us to implement other behaviour under the same api
for StackTracePrinter, for example, MarkupStackTracePrinter.

### CODE REVIEW

File: compiler-rt/lib/sanitizer_common/sanitizer_stacktrace_printer.cpp [hunk 1]
```diff
@@ -12,13 +12,28 @@
...
 
+#include "sanitizer_common.h"
 #include "sanitizer_file.h"
 #include "sanitizer_flags.h"
 #include "sanitizer_fuchsia.h"
 
 namespace __sanitizer {
 
-const char *StripFunctionName(const char *function) {
+StackTracePrinter *StackTracePrinter::stacktrace_printer_;
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_stacktrace_printer.cpp:22]: Please remove these from header and just use

`static StackTracePrinter *stacktrace_printer;`

Even better GetOrInit  function local one

File: compiler-rt/lib/sanitizer_common/tests/sanitizer_stacktrace_printer_test.cpp [hunk 2]
```diff
@@ -16,57 +16,64 @@
...
-  RenderSourceLocation(&str, "/dir/file.cc", 10, 5, true, "/dir/");
+  printer->RenderSourceLocation(&str, "/dir/file.cc", 10, 5, true, "/dir/");
   EXPECT_STREQ("file.cc(10,5)", str.data());
 }
 
-TEST(SanitizerStacktracePrinter, RenderModuleLocation) {
+TEST(FormattedStackTracePrinter, RenderModuleLocation) {
   InternalScopedString str;
-  RenderModuleLocation(&str, "/dir/exe", 0x123, kModuleArchUnknown, "");
+  FormattedStackTracePrinter *printer = new FormattedStackTracePrinter();
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/tests/sanitizer_stacktrace_printer_test.cpp:57]: we don't need to leak here 

```
class TestFormattedStackTracePrinter : public FormattedStackTracePrinter {
public:
 ~ TestFormattedStackTracePrinter()
}
```

And you can declare them on stack

**Review by vitalybuka — APPROVED** [review]
With 2 suggestions

File: compiler-rt/lib/sanitizer_common/tests/sanitizer_stacktrace_printer_test.cpp (hunk 2 above)
Comment (avillega) [reply compiler-rt/lib/sanitizer_common/tests/sanitizer_stacktrace_printer_test.cpp:57]: Thanks for this tip, I battled a lot with this code for this exact reason.

File: compiler-rt/lib/sanitizer_common/sanitizer_stacktrace_printer.cpp (hunk 1 above)
Comment (avillega) [reply compiler-rt/lib/sanitizer_common/sanitizer_stacktrace_printer.cpp]: made both static local to the function

**Pull request merged**